### PR TITLE
Allow empty request path in AbstractHttpRequestBuilder

### PR DESCRIPTION
**Pull request by eisig**
Motivation:

As describe in  https://github.com/line/armeria/issues/3824.

A client build with ` WebClient.of("http://localhost/test/issues")` 
would be able to send request to  both `http://localhost/test/issues` and `http://localhost/test/issues/1`

Result:
- You can now specify an empty path for the request path.

### CODE REVIEW

File: core/src/test/java/com/linecorp/armeria/common/HttpRequestBuilderTest.java [hunk 1]
```diff
@@ -78,10 +78,15 @@ void buildSimple() {
...
-                .isInstanceOf(IllegalArgumentException.class)
-                .hasMessageContaining("empty");
+    void shouldAllowEmptyPath() {
+        final HttpRequest request = HttpRequest.builder().get("").build();
+        assertThat(request.path()).isEqualTo("");
+    }
+
+    @Test
+    void buildEmptyPathWithQueryParams() {
+        final HttpRequest request = HttpRequest.builder().get("").queryParam("foo","bar").build();
```
Comment (minwoox) [inline core/src/test/java/com/linecorp/armeria/common/HttpRequestBuilderTest.java:88]: nit: 
```suggestion
        final HttpRequest request = HttpRequest.builder().get("").queryParam("foo", "bar").build();
```

**Review by minwoox — APPROVED** [review]
Thanks! 😄

**Review by ikhoon — APPROVED** [review]
LGTM!

**Review by jrhee17 — APPROVED** [review]
Thank you!

**Comment by minwoox** [comment]
Thanks, @eisig! 👍

**Pull request merged**